bin/node/cli/src/service.rs

```diff
@@ -96,8 +96,22 @@ macro_rules! new_full_start {
 				import_setup = Some((block_import, grandpa_link, babe_link));
 				Ok(import_queue)
 			})?
-			.with_rpc_extensions(|client, pool, _backend, fetcher, _remote_blockchain| -> Result<RpcExtension, _> {
-				Ok(node_rpc::create(client, pool, node_rpc::LightDeps::none(fetcher)))
+			.with_rpc_extensions(|client, pool, _backend, _fetcher, _remote_blockchain, select_chain, keystore| ->
+				Result<RpcExtension, _>
+			{
+				let babe_link = import_setup.as_ref().map(|s| &s.2)
+					.expect("Link is present or set up failed; qed.");
+				let deps = node_rpc::FullDeps {
+					client,
+					pool,
+					select_chain: select_chain.expect("SelectChain is present or set up failed; qed."),
+					babe: node_rpc::BabeDeps {
+						keystore,
+						babe_config: sc_consensus_babe::BabeLink::config(babe_link).clone(),
+						shared_epoch_changes: sc_consensus_babe::BabeLink::epoch_changes(babe_link).clone()
```

Comment:
```suggestion
						shared_epoch_changes: sc_consensus_babe::BabeLink::epoch_changes(babe_link),
```

Reply:
`BabeLink::epoch_changes` returns a reference to `SharedEpochChanges`